File: app/src/main/java/com/riders/thelab/ui/contacts/ContactDetailActivity.java
package com.riders.thelab.ui.contacts;

import android.annotation.SuppressLint;
import android.content.Context;
import android.os.Bundle;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.appcompat.widget.Toolbar;
import androidx.core.view.ViewCompat;

import com.google.android.material.appbar.AppBarLayout;
import com.google.android.material.appbar.CollapsingToolbarLayout;
import com.riders.thelab.R;
import com.riders.thelab.ui.base.SimpleActivity;

import butterknife.BindView;
import butterknife.ButterKnife;
import timber.log.Timber;

@SuppressLint("NonConstantResourceId")
public class ContactDetailActivity extends SimpleActivity {

    private Context mContext;

    @BindView(R.id.app_bar_layout)
    AppBarLayout appBarLayout;
    @BindView(R.id.contact_detail_collapsing_toolbar)
    CollapsingToolbarLayout mCollapsingToolbar;
    @BindView(R.id.contact_detail_toolbar)
    Toolbar toolbar;
    @BindView(R.id.iv_image_detail_toolbar)
    ImageView mImageDetailToolbar;
    @BindView(R.id.tv_name_detail)
    TextView mNameTextView;
    @BindView(R.id.tv_surname_detail)
    TextView mSurnameTextView;
    @BindView(R.id.tv_email_detail)
    TextView mEmailTextView;

    public static final String CONTACT_NAME = "contact_name";
    public static final String CONTACT_SURNAME = "contact_surname";
    public static final String CONTACT_EMAIL = "contact_email";
    public static final String CONTACT_IMAGE = "contact_image";

    String itemNameDetail;
    String itemSurnameDetail;
    String itemEmailDetail;
    String itemImage;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_contacts_detail);

        mContext = this;

        ButterKnife.bind(this);

        getBundle();

        ViewCompat.setTransitionName(findViewById(R.id.app_bar_layout), String.valueOf(R.drawable.image5));
        supportPostponeEnterTransition();

        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setDisplayShowHomeEnabled(true);

        mCollapsingToolbar.setTitle(itemNameDetail);
        mCollapsingToolbar.setExpandedTitleColor(getResources().getColor(android.R.color.transparent));
//        getSupportActionBar().setTitle(itemNameDetail);

        appBarLayout.setExpanded(true);

        appBarLayout.addOnOffsetChangedListener(new AppBarLayout.OnOffsetChangedListener() {
            boolean isShow = false;
            int scrollRange = -1;

            @Override
            public void onOffsetChanged(AppBarLayout appBarLayout, int verticalOffset) {
                if (scrollRange == -1) {
                    scrollRange = appBarLayout.getTotalScrollRange();
                }
                if (scrollRange + verticalOffset == 0) {
                    mCollapsingToolbar.setTitle(itemNameDetail);
                    isShow = true;
                } else if (isShow) {
                    mCollapsingToolbar.setTitle(" ");
                    isShow = false;
                }
            }
        });
    }

    private void getBundle() {
        Bundle bundle = getIntent().getExtras();

        if (null == bundle) {
            Timber.e("bundle is null");
            return;
        }

        itemNameDetail = bundle.getString(CONTACT_NAME);
        itemEmailDetail = bundle.getString(CONTACT_EMAIL);
        itemImage = bundle.getString(CONTACT_IMAGE);

        setViews();
    }

    private void setViews() {
        mNameTextView.setText(itemNameDetail);
        mEmailTextView.setText(itemEmailDetail);
    }
}

File: app/src/main/java/com/riders/thelab/ui/contacts/ContactDetailActivity.kt
package com.riders.thelab.ui.contacts

import android.content.Context
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import androidx.core.view.ViewCompat
import com.google.android.material.appbar.AppBarLayout
import com.google.android.material.appbar.AppBarLayout.OnOffsetChangedListener
import com.riders.thelab.R
import com.riders.thelab.databinding.ActivityContactsDetailBinding
import dagger.hilt.android.AndroidEntryPoint
import timber.log.Timber

@AndroidEntryPoint
class ContactDetailActivity : AppCompatActivity() {

    companion object {
        const val CONTACT_NAME = "contact_name"
        const val CONTACT_SURNAME = "contact_surname"
        const val CONTACT_EMAIL = "contact_email"
        const val CONTACT_IMAGE = "contact_image"
    }

    private var mContext: Context? = null

    private var _viewBinding: ActivityContactsDetailBinding? = null

    // This property is only valid between onCreateView and
    // onDestroyView.
    private val binding get() = _viewBinding!!

    var itemNameDetail: String? = null
    var itemSurnameDetail: String? = null
    private var itemEmailDetail: String? = null
    private var itemImage: String? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        _viewBinding = ActivityContactsDetailBinding.inflate(layoutInflater)
        setContentView(binding.root)

        mContext = this

        getBundle()

        ViewCompat.setTransitionName(binding.appBarLayout, "icon")
        supportPostponeEnterTransition()

        /*setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setDisplayShowHomeEnabled(true);*/
        setSupportActionBar(binding.contactDetailToolbar)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)

        binding.contactDetailCollapsingToolbar.title = itemNameDetail
        binding.contactDetailCollapsingToolbar.setExpandedTitleColor(
            ContextCompat.getColor(
                this,
                R.color.transparent
            )
        )
        //        getSupportActionBar().setTitle(itemNameDetail);

        binding.appBarLayout.setExpanded(true)
        binding.appBarLayout.addOnOffsetChangedListener(object : OnOffsetChangedListener {
            var isShow = false
            var scrollRange = -1
            override fun onOffsetChanged(appBarLayout: AppBarLayout, verticalOffset: Int) {
                if (scrollRange == -1) {
                    scrollRange = appBarLayout.totalScrollRange
                }
                if (scrollRange + verticalOffset == 0) {
                    binding.contactDetailCollapsingToolbar.title = itemNameDetail
                    isShow = true
                } else if (isShow) {
                    binding.contactDetailCollapsingToolbar.title = " "
                    isShow = false
                }
            }
        })
    }

    override fun onDestroy() {
        super.onDestroy()
        Timber.e("onDestroy()")
        _viewBinding = null
    }

    private fun getBundle() {
        val bundle = intent.extras
        if (null == bundle) {
            Timber.e("bundle is null")
            return
        }
        itemNameDetail = bundle.getString(CONTACT_NAME)
        itemEmailDetail = bundle.getString(CONTACT_EMAIL)
        itemImage = bundle.getString(CONTACT_IMAGE)
        setViews()
    }

    private fun setViews() {
        binding.tvNameDetail.text = itemNameDetail
        binding.tvEmailDetail.text = itemEmailDetail
    }
}